src/gui/src/drcWidget.cpp

```diff
@@ -0,0 +1,538 @@
+      odb::dbTechLayer* layer = tech->findLayer(base_match[2].str().c_str());
+
+      odb::Rect rect;
+      if (std::regex_match(bbox, base_match, bbox_corners)) {
+        rect.set_xlo(std::stod(base_match[1].str()) * block_->getDbUnitsPerMicron());
+        rect.set_ylo(std::stod(base_match[2].str()) * block_->getDbUnitsPerMicron());
+        rect.set_xhi(std::stod(base_match[3].str()) * block_->getDbUnitsPerMicron());
+        rect.set_yhi(std::stod(base_match[4].str()) * block_->getDbUnitsPerMicron());
+      }
+
+      std::vector<std::any> srcs_list;
+      std::stringstream srcs_stream(sources);
+      std::string single_source;
+
+      // split sources list
+      while(getline(srcs_stream, single_source, ' ')) {
+        if (single_source.empty()) {
+          continue;
+        }
+
+        auto pin_loc = single_source.find("PIN/");
+        if (pin_loc != std::string::npos) {
+          std::string pin_name = single_source.substr(pin_loc + 1);
+          if (pin_name != "OBS") {
+            // bterm
+            odb::dbBTerm* bterm = block_->findBTerm(pin_name.c_str());
+            if (bterm != nullptr) {
+              srcs_list.push_back(bterm);
+            }
+          }
+        } else {
+          auto inst_loc = single_source.find("/");
+          if (inst_loc != std::string::npos) {
+            std::string inst_name = single_source.substr(0, inst_loc);
+            std::string pin_name = single_source.substr(inst_loc + 1);
+            // instance
+            odb::dbInst* inst = block_->findInst(inst_name.c_str());
+            if (inst != nullptr) {
+              if (pin_name == "OBS") {
+                // just add instance
+                srcs_list.push_back(inst);
+              } else {
+                odb::dbITerm* iterm = nullptr;
+                for (auto itrm : inst->getITerms()) {
+                  if (itrm->getMTerm()->getName() == pin_name) {
+                    iterm = itrm;
+                  }
+                }
+                if (iterm != nullptr) {
+                  // add iterm
+                  srcs_list.push_back(iterm);
+                } else {
+                  // add inst
+                  srcs_list.push_back(inst);
+                }
+              }
+            }
+          } else {
+            // net
+            odb::dbNet* net = block_->findNet(single_source.c_str());
+            if (net != nullptr) {
+              srcs_list.push_back(net);
+            }
+          }
+        }
+      }
+
+      std::vector<DRCViolation::DRCShape> shapes({
+        QRect{rect.xMin(), rect.yMin(), static_cast<int>(rect.dx()), static_cast<int>(rect.dy())}});
+      violations_.push_back(std::make_unique<DRCViolation>(
+        type,
+        srcs_list,
+        shapes,
+        layer,
+        sources));
+    }
+  }
+
+  report.close();
+}
+
+void DRCWidget::loadASCIIReport(const QString& filename)
+{
+  logger_->error(utl::GUI, 31, "ASCII databases not supported.");
```

Comment:
Are you planning to do this soon?

Reply:
Yes, this is just a placeholder since I wanted to get some feedback about the UI before plowing forward. Also there was the question about reverse engineering of that format to be resolved, since the format is pretty simple it's not much effort.